src/plugin/plugin.ts

```diff
@@ -843,6 +848,55 @@ class Purchases {
   ) {
     window.cordova.exec(callback, null, PLUGIN_NAME, "checkTrialOrIntroductoryPriceEligibility", [productIdentifiers]);
   }
+  /**
+   * Sets a function to be called on purchases initiated on the Apple App Store. This is only used in iOS.
+   * @param {ShouldPurchasePromoProductListener} shouldPurchasePromoProductListener Called when a user initiates a
+   * promotional in-app purchase from the App Store. If your app is able to handle a purchase at the current time, run
+   * the deferredPurchase function. If the app is not in a state to make a purchase: cache the deferredPurchase, then
+   * call the deferredPurchase when the app is ready to make the promotional purchase.
+   * If the purchase should never be made, you don't need to ever call the deferredPurchase and the app will not
+   * proceed with promotional purchases.
+   */
+  public static addShouldPurchasePromoProductListener(shouldPurchasePromoProductListener: ShouldPurchasePromoProductListener): void { 
+    if (typeof shouldPurchasePromoProductListener !== "function") {
+      throw new Error("addPurchaserInfoUpdateListener needs a function");
```

Comment:
Oops, fixing

Reply:
Cool. I commented on my review on this

Reply:
@vegaro https://github.com/RevenueCat/react-native-purchases/pull/98/files#diff-f41e9d04a45c83f3b6f6e630f10117feR546 LOL 😂 copy/paste kings over here. I'll make a PR for that one as well.